irohad/torii/config.hpp

```diff
@@ -15,30 +15,21 @@
  * limitations under the License.
  */
 
-#ifndef IROHA_MAIN_CLI_LEDGER_CREATE_HPP_
-#define IROHA_MAIN_CLI_LEDGER_CREATE_HPP_
+#ifndef IROHA_TORII_CONFIG_HPP_
+#define IROHA_TORII_CONFIG_HPP_
 
-#include "main/cli/config.hpp"
+#include "cli/common.hpp"
 
 namespace iroha {
-  namespace cli {
-    namespace handler {
-      namespace ledger {
-
-        void create(
-            config::Postgres *pg,
-            config::Redis *rd,
-            config::BlockStorage *bs,
-            const std::string& genesis_content
-        ) {
-          // TODO: implement
-          // raw block inserter?
-          printf("genesis block content: \n%s", genesis_content.c_str());
-        }
-
-      }
+  namespace torii {
+    namespace config {
+      /**
+       * @struct Config::Torii
+       * @brief Config for torii gateway.
+       */
+      class Torii : public iroha::config::Service {};
```

Comment:
Could you explain your class hierarchy?

Reply:
Service is something that requires connection details: host, port.
Service with authentication is a Service (has host, port) which has auth details, such as username and password.

Torii here is a service. As well as Redis. Postgres is AuthService.

Reply:
gotcha, thx!